roles/ddns_updater/defaults/main.yml

```diff
@@ -0,0 +1,45 @@
+---
+# enable the role
+ddns_updater_enabled: false
+
+# docker image used
+ddns_updater_docker_image:  qmcgaw/ddns-updater:latest
+
+# Data directory for config file
+ddns_updater_data_directory: "{{ docker_home }}/ddns_updater"
+
+# enable in order to make web ui reachable from external
+ddns_updater_available_externally: "false"
+
+# Port for the web ui
+ddns_updater_port: 8202
+ddns_updater_hostname: "ddns"
+
+# User to run the container with
+ddns_updater_user_id: "0"
+
+# Token to be used to send messages via gotify
+# ddns_updater_gotify_token: "ABCDEFGHIJKLMOP"
+
+# url of gotify server
+gotify_url: "gotify.{{ ansible_nas_domain }}"
+
+# dict to pass in additional environment variables
+# available in the docker images (https://github.com/qdm12/ddns-updater#environment-variables)
+ddns_updater_extra_env: {}
+
+# ddns_updater_config describes the content of the config.json
+# used to configure the updater
+# see https://github.com/qdm12/ddns-updater#configuration
+#
+# using cloudflare should work with a config like this:
+#
+#ddns_updater_config:
+#  settings:
```

Comment:
Seeing how this works, I think it's a little over-engineered. We're expecting users to mentally map a json config example in the container docs to YAML, which then gets mapped back to JSON and written out to a template file when the role is run. It would be easier and less brain-hurty to simply accept a json config block and write that out to the config file:

```
ddns_updater_config: >
{
  "settings": [
    {
      "provider": "cloudflare",
      "zone_identifier": "some id",
      "domain": "domain.com",
      "host": "@",
      "ttl": 600,
      "token": "yourtoken",
      "ip_version": "ipv4"
    }
  ]
}

```

Reply:
of course, thats way better. I updated that part